### PR TITLE
Clarified config documentation regarding preference of multiple sources

### PR DESCRIPTION
**Pull request by alneberg**
I had some difficulties understanding the order of preference for configurations in the current documentation. @ewels pointed me to this archived documentation which helped me:
https://github.com/SciLifeLab/NGI-NextflowDocs#configuration-loading

These changes are based on those. Maybe someone more experienced should check whether this is actually correct.

Also, this might a subjective preference of the documentation so feel free to close this PR or ask me to change it further.

### CODE REVIEW

File: docs/config.rst [hunk 1]
```diff
@@ -7,18 +7,19 @@ Configuration
...
-
-.. note:: It's worth noting that by doing this, the files ``nextflow.config`` and ``$HOME/.nextflow/config`` are not
-  ignored and they are merged as explained above.
+When a pipeline script is launched Nextflow looks for configurations in multiple locations. Since multiple files can
+contain conflicting settings the sources are ranked to decide which setting to come in play. The possible sources for
+configuration values are listed below with the highest preference at the top:
+
+1. Command line individual parameter (``--something value``)
+2. Command line specified config file (``-c my_config``)
+3. A file named ``nextflow.config`` in the current directory
```
Comment (ewels) [inline docs/config.rst]: current _working_ directory may be fractionally clearer...

File: docs/config.rst [hunk 2]
```diff
@@ -7,18 +7,19 @@ Configuration
...
+4. A file named ``nextflow.config`` in the script base directory (when it is not the same as the current directory)
+5. The file ``$HOME/.nextflow/config``
```
Comment (ewels) [inline docs/config.rst]: Is this right @pditommaso? That the pipeline default config trumps the user's home directory config? I would have expected 4. and 5. to be the other way around here.

I think maybe it is right and we've discussed this before... My memory's not what it used to be.. 🤔

File: docs/config.rst (hunk 1 above)
Comment (ewels) [reply docs/config.rst:16]:
```suggestion
3. A file named ``nextflow.config`` in the current working directory
```

File: docs/config.rst (hunk 2 above)
Comment (pditommaso) [reply docs/config.rst:18]: Yes, it does. That's useful for user default settings, like mail SMTP server, etc

Comment (ewels) [reply docs/config.rst:18]: Ok great! In that case I think that this PR is ready to go?

Comment (ewels) [reply docs/config.rst:18]: And to confirm: this reads that any settings in the pipeline `nextflow.config` has priority over a config file in the home directory?

This still surprises me as email / other parameters are typically initialised as `false` in the pipeline `nextflow.config` file. Which would mean that they would effectively be impossible to have defaults for in the home directory config.

Comment (pditommaso) [reply docs/config.rst:18]: > this reads that any settings in the pipeline nextflow.config has priority over a config file in the home directory

Yes, the logic is the local overrides the common defaults

> other parameters are typically initialised as false in the pipeline nextflow.config file

the pipeline config, can still be overridden by the local config, tho I get what you are suggesting, that the pipeline should be the root, then "home" config, etc

**Review by ewels — COMMENTED** [review]
Might be nice to add in `-params-file` as I don't think that this option existed when the PR was started 😅

**Comment by pditommaso** [comment]
Ok, merged. I've added the reference to `-params-file` option as @ewels suggested 👉  1ae3855c. 

Thanks for your patience 😄

**Comment by ewels** [comment]
2 years, 2 months, 20 days - not bad! 😆 

Thank you!

**Comment by pditommaso** [comment]
Trolling! 😆

**Comment by ewels** [comment]
It's usually me being slow to merge PRs so I enjoyed the experience! 😆

**Pull request closed without merging**